internal/app/machined/pkg/controllers/files/udev_rule_file.go

```diff
@@ -0,0 +1,118 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package files
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"go.uber.org/zap"
+
+	runtimetalos "github.com/siderolabs/talos/internal/app/machined/pkg/runtime"
+	"github.com/siderolabs/talos/pkg/machinery/resources/files"
+)
+
+// UdevRuleFileController is a controller for UdevRule files.
+type UdevRuleFileController struct {
+	V1Alpha1Mode  runtimetalos.Mode
+	UdevRulesFile string
+	CommandRunner func(ctx context.Context, name string, args ...string) (string, error)
+}
+
+// Name implements controller.Controller interface.
+func (ctrl *UdevRuleFileController) Name() string {
+	return "files.UdevRuleFileController"
+}
+
+// Inputs implements controller.Controller interface.
+func (ctrl *UdevRuleFileController) Inputs() []controller.Input {
+	return []controller.Input{
+		{
+			Namespace: files.NamespaceName,
+			Type:      files.UdevRuleType,
+			Kind:      controller.InputWeak,
+		},
+	}
+}
+
+// Outputs implements controller.Controller interface.
+func (ctrl *UdevRuleFileController) Outputs() []controller.Output {
+	return []controller.Output{
+		{
+			Type: files.UdevRuleStatusType,
+			Kind: controller.OutputExclusive,
+		},
+	}
+}
+
+// Run implements controller.Controller interface.
+//
+// nolint:gocyclo
+func (ctrl *UdevRuleFileController) Run(ctx context.Context, r controller.Runtime, logger *zap.Logger) error {
```

Comment:
I'm confused on how to handle this, currently this behaves like the previous implementation, write all rules to a single file and reload udev. Another option could be have individual file for each rule, but then I am confused on how to prefix file names (priority order based on numeric prefix) and the cleanup process, do we only delete the file the controller created or all other files under the folder (since extensions also bring in udev rules into the same directory)